calico/network-policy/policy-rules/log-rules.mdx

```diff
@@ -0,0 +1,206 @@
+    1. **Explicit Allow Rule:** Always pair a log action with an explicit `Allow` rule immediately after it. Otherwise, traffic may be denied due to the tier's implicit default action.  
+    1. **Comprehensive Protection:** Ensure both namespaced and non-namespaced resources are covered by enabling `hostEndpoint` for protecting your host in addition to Pods and workloads.  
+
+By strategically placing log actions at the bottom of your policy tiers, you can capture unhandled traffic and identify gaps in your policy configuration. Analyze the logged traffic to gather insights about source, destination, and protocol, and use this data to create targeted policies. Once implemented, validate by checking if the corresponding traffic is no longer logged, indicating the new policies are effective.  
+
+
+### Logging an unprotected cluster 
+
+Kubernetes' default behavior permits all traffic, making early security implementation essential. $[prodname]'s `GlobalNetworkPolicy` with the `Log` action is a powerful tool for monitoring cluster traffic. It allows you to observe, analyze, and secure traffic flows. A single `GlobalNetworkPolicy` can be used to log all traffic across the cluster, providing comprehensive insights for your security strategy.  
+
+
+To log all traffic across the entire cluster, define a GlobalNetworkPolicy:
+```yaml
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+    name: log-everything-in-cluster
+spec:
+    order: 100000
+    tier: default
+    types:
+    - Ingress
+    - Egress
+    egress:
+    - action: Log
+    ingress:
+    - action: Log
+```
+
+### Logging in a namespace
+
+To monitor or troubleshoot traffic within a namespace, use a NetworkPolicy. Apply the policy with the highest priority to capture traffic not handled by existing policies. This provides visibility into unhandled traffic patterns specific to that namespace.
+
+To log all traffic within a specific namespace, create a NetworkPolicy like this:
+```yaml
+apiVersion: projectcalico.org/v3
+kind: NetworkPolicy
+metadata:
+  name: log-everything-in-ns
+  namespace: <YOUR-NAMESPACE>
+spec:
+  order: 100001
+  tier: default
+  types:
+  - Ingress
+  - Egress
+  egress:
+  - action: Log
+  ingress:
+  - action: Log
+```
+
+### Logging all traffic in a cluster
+
+By default, $[prodname] policies apply only to namespaced resources (e.g., Pods, ServiceAccounts). To achieve total protection, enable hostEndpoint support to include non-namespaced resources and host nodes.
+
+To log all traffic involving host endpoints, enable auto-creation of host endpoints:
+```bash
+kubectl patch kubecontrollersconfiguration default --type=merge --patch='{"spec": {"controllers": {"node": {"hostEndpoint": {"autoCreate": "Enabled"}}}}}'
+```
+
+Please ensure you disable policy logging by removing the Log policies once you have finalized your environment security, as it can impact cluster performance.
```

Comment:
This seems very important. You should make the full sequence clear from the beginning. 

I wonder whether we should have a section somewhere about lifecycle for log actions. Is it best to set and forget? Only run for a set period? What about the time range, data allowances, and so on? Even if this isn't the place for all that detail, it may be worth mentioning and linking from Additional resources.

Reply:
As we discussed in the 1on1, a new user should not try to enable the host endpoint right at the bat since it may break their cluster if they don't know what they are doing. Moving this step any higher would create a lot of support cases like I enabled logs and it broke my cluster.
In my opinion, the flow of discovery should be as it is written on the page.

Reply:
I'm not suggesting a change to the actual workflow. What I'm saying is that it would be better for users if you first explain what the workflow is, and then show them how to do it.

In particular, I'm interested in the idea that you suggest here at the end. It seems that, at least in some cases, log rules are best used on a temporary basis. You set them, figure something else, and then remove them. That's an idea that would be good to present up at the top.